main.go

```diff
@@ -37,6 +37,12 @@ func init() {
 }
 
 func startMetricPartitioningRoutine(serviceConfig config.Config, service service.ProxyService, serviceLogger logging.ServiceLogger) <-chan error {
+	if !serviceConfig.MetricDatabaseEnabled {
```

Comment:
I think these checks would be more clear in the caller `main()` to conditionally call them instead of returning within the functions -- eg only run those 3 background routines if `serviceConfig.MetricDatabaseEnabled`

Reply:
Updated. Did those initially as it was done for one routine in this way, but make sense as it disables all 3 of those